.travis.yml

```diff
@@ -13,6 +13,8 @@ before_install:
   - gem install bundler -v 2.0.1
   - bundle update
   - unset _JAVA_OPTIONS
+  - mkdir ./tmp
```

Comment:
@abunashir can we not create the `./tmp` here but use `$TMPDIR` given by the system?

Reply:
(see below comments)

Reply:
@ronaldtse: What are the benefits of using a shared system's temp directory over project-specific `tmp` directory?

We are only using this directory for the test purpose, and to me, it ensures we can only write to this tmp directly and no other program's tmp file can possibly cause a failing test here and we also ignore this the git, so what are disadvantage we are considering here?

Reply:
The disadvantage is that we now have a `.travis.yml` that specifically relates to these two specs that use this directory. I'd much rather to have that directory done inside the spec. (We want the travis files to be identical across all gems).

Reply:
Okay, what about we remove it from the travis configuration and add RSpec configure block that will only run once before the tests and create the temp directory (if none exists)?

```ruby
config.before(:suite) do
  # create non-existance tmp directory
end
```